### ISSUE TITLE
SAM doesn't seem to be doing well

### ISSUE THREAD

**Issue by AlexLiu14** [davda54/sam#87]
Thanks for providing a nice code for SAM. However, I think I'm having some problems. My model seemed to perform even worse when SAM was added, dropping from 90% accuracy to 60% accuracy. The model is resnet50, and Loss is defined by a class, loss_computer, that adds some computation, using the loss function CrossEntropyLoss().Here is my code.
`
        for batch_idx, batch in enumerate(prog_bar_loader):

            batch = tuple(t.cuda() for t in batch)
            x = batch[0]
            y = batch[1]
            g = batch[2]
            
            outputs = model(x)
            if is_training:
                enable_running_stats(model)
            loss_main = loss_computer.loss(outputs, y, g, is_training)

            if is_training:
                loss_main.backward()
                optimizer.first_step(zero_grad=True)
                # second forward-backward pass
                disable_running_stats(model)
                loss_computer.loss(model(x), y, g, is_training).backward()  # make sure to do a full forward pass
                optimizer.second_step(zero_grad=True)
`
Please help me see if I using SAM right. I'm glad to take your any advice. Thank you.

**Comment by rlczddl** [comment]
I found that a large learning rate is needed.
